src/js/components/picklist/picklist.scss

```diff
@@ -54,6 +54,11 @@
   .icon:first-child {
     margin-right: 8px;
   }
+
+  .icon.fa-check {
+    margin-left: auto;
+    margin-right: 0;
+  }
```

Comment:
This fixes the right alignment issue with the `Filter by Date…` picklist checkmarks. Should be able to remove this when the component is configured to use the `isCheckable` option.